src/map/homunculus.cpp

```diff
@@ -1483,216 +1437,555 @@ uint8 hom_get_intimacy_grade(struct homun_data *hd) {
+			}
+
+			if (this->nodeExists(skill, "RequiredLevel")) {
+				uint16 level;
+
+				if (!this->asUInt16(skill, "RequiredLevel", level))
+					return 0;
+
+				uint16 config_max = battle_config.hom_max_level;
+
+				if ((hom_class2type(class_id) == HT_S))
+					config_max = battle_config.hom_S_max_level;
+
+				if (level > config_max) {
+					this->invalidWarning(skill["RequiredLevel"], "Homunculus Required Skill level %u exceeds maximum level %u, capping.\n", level, config_max);
+					level = config_max;
+				}
+
+				entry.need_level = static_cast<uint8>(level);
+			} else {
+				if (!exists)
+					entry.need_level = 0;
+			}
+
+			if (this->nodeExists(skill, "RequiredIntimacy")) {
+				uint32 intimacy;
+
+				if (!this->asUInt32(skill, "RequiredIntimacy", intimacy))
+					return 0;
+
+				if (intimacy > 100000) {
+					this->invalidWarning(skill["RequiredIntimacy"], "Homunculus Required Intimacy %u exceeds maximum intimacy 100000, capping.\n", intimacy);
+					intimacy = 100000;
+				}
+
+				entry.intimacy = intimacy;
+			} else {
+				if (!exists)
+					entry.intimacy = 0;
+			}
+
+			if (this->nodeExists(skill, "Required")) {
+				const ryml::NodeRef &required = skill["Required"];
+
+				for (const ryml::NodeRef &prereqskill : required) {
+					uint16 skill_id = 0, skill_lv = 0;
+
+					if (this->nodeExists(prereqskill, "Skill")) {
+						std::string skill_name;
+
+						if (!this->asString(prereqskill, "Skill", skill_name))
+							return 0;
+
+						if (skill_name2id(skill_name.c_str()) == 0) {
+							this->invalidWarning(prereqskill["Skill"], "Invalid homunculus skill %s, skipping.\n", skill_name.c_str());
+							return 0;
+						}
+
+						skill_id = skill_name2id(skill_name.c_str());
+
+						if (skill_id == 0) {
+							this->invalidWarning(prereqskill["Skill"], "Invalid homunculus skill %s, skipping.\n", skill_name.c_str());
+							return 0;
+						}
+
+						if (!SKILL_CHK_HOMUN(skill_id)) {
+							this->invalidWarning(prereqskill["Skill"], "Homunculus skill %s (%u) is out of the homunculus skill range [%u-%u], skipping.\n", skill_name.c_str(), skill_id, HM_SKILLBASE, HM_SKILLBASE + MAX_HOMUNSKILL - 1);
+							return 0;
+						}
+					}
+
+					if (this->nodeExists(prereqskill, "Clear")) {
+						bool found = false;
+
+						for (auto &skit : hom->skill_tree) {
+							auto it = skit.need.begin();
+
+							while (it != skit.need.end()) {
+								if (it->first == skill_id) { // Skill found, remove it from the skill tree.
+									it = skit.need.erase(it);
+									found = true;
+								} else {
+									it++;
+								}
+							}
+						}
+
+						if (!found)
+							this->invalidWarning(prereqskill["Clear"], "Failed to remove nonexistent prerequisite skill %s from homunuculus %s.\n", skill_db.find(skill_id)->name, class_name.c_str());
+						continue;
+					}
+
+					if (this->nodeExists(prereqskill, "Level")) {
+						if (!this->asUInt16(prereqskill, "Level", skill_lv))
+							return 0;
+					}
```

Comment:
Not sure if level should not be "1" in the else case.
We could think about using level = 0 instead of clear for removing too.